libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.ts

```diff
@@ -212,212 +243,184 @@ export class SkyDateRangePickerComponent
-    this.#_valueOrDefault = value;
-    this.#updateSelectedCalculator();
-  }
+  public ngOnInit(): void {
+    if (this.#labelTextRequiredSvc) {
+      this.#labelTextRequiredSvc.validateLabelText(this.label);
 
-  get #valueOrDefault(): SkyDateRangeCalculation | undefined {
-    return this.#_valueOrDefault;
+      if (!this.label) {
+        this.display = 'none';
+      }
+    }
   }
 
-  readonly #labelTextRequired = inject(SkyFormFieldLabelTextRequiredService, {
-    optional: true,
-  });
-
-  readonly #appFormatter = inject(SkyAppFormat);
-  #control: AbstractControl | undefined;
-  #preferredShortDateFormat: string | undefined;
-  #ngUnsubscribe = new Subject<void>();
-  readonly #resourceSvc = inject(SkyLibResourcesService);
-
-  #_calculatorIds: SkyDateRangeCalculatorId[] = [
-    SkyDateRangeCalculatorId.AnyTime,
-    SkyDateRangeCalculatorId.Before,
-    SkyDateRangeCalculatorId.After,
-    SkyDateRangeCalculatorId.SpecificRange,
-    SkyDateRangeCalculatorId.Yesterday,
-    SkyDateRangeCalculatorId.Today,
-    SkyDateRangeCalculatorId.Tomorrow,
-    SkyDateRangeCalculatorId.LastWeek,
-    SkyDateRangeCalculatorId.ThisWeek,
-    SkyDateRangeCalculatorId.NextWeek,
-    SkyDateRangeCalculatorId.LastMonth,
-    SkyDateRangeCalculatorId.ThisMonth,
-    SkyDateRangeCalculatorId.NextMonth,
-    SkyDateRangeCalculatorId.LastQuarter,
-    SkyDateRangeCalculatorId.ThisQuarter,
-    SkyDateRangeCalculatorId.NextQuarter,
-    SkyDateRangeCalculatorId.LastCalendarYear,
-    SkyDateRangeCalculatorId.ThisCalendarYear,
-    SkyDateRangeCalculatorId.NextCalendarYear,
-    SkyDateRangeCalculatorId.LastFiscalYear,
-    SkyDateRangeCalculatorId.ThisFiscalYear,
-    SkyDateRangeCalculatorId.NextFiscalYear,
-  ];
-  #_dateFormat: string | undefined;
-  #_disabled: boolean | undefined = false;
-  #_valueOrDefault: SkyDateRangeCalculation | undefined;
-
-  #changeDetector: ChangeDetectorRef;
-  #dateRangeService: SkyDateRangeService;
-  #formBuilder: UntypedFormBuilder;
-  #localeProvider: SkyAppLocaleProvider;
-  #ngZone: NgZone;
-
-  constructor(
-    changeDetector: ChangeDetectorRef,
-    dateRangeService: SkyDateRangeService,
-    formBuilder: UntypedFormBuilder,
-    localeProvider: SkyAppLocaleProvider,
-    ngZone: NgZone,
-    @Optional() themeSvc?: SkyThemeService,
-  ) {
-    this.#changeDetector = changeDetector;
-    this.#dateRangeService = dateRangeService;
-    this.#formBuilder = formBuilder;
-    this.#localeProvider = localeProvider;
-    this.#ngZone = ngZone;
-
-    this.dateFormatOrDefault = this.dateFormat;
-
-    this.#localeProvider
-      .getLocaleInfo()
-      .pipe(takeUntil(this.#ngUnsubscribe))
-      .subscribe((localeInfo) => {
-        SkyDateFormatter.setLocale(localeInfo.locale);
-        this.#preferredShortDateFormat =
-          SkyDateFormatter.getPreferredShortDateFormat();
-        this.dateFormatOrDefault =
-          this.dateFormat || this.#preferredShortDateFormat;
```

Comment:
Responding to locale changes is being handled by the datepicker, so this duplicate logic can be removed.